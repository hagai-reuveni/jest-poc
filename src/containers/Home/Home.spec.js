import React from 'react';
import { Home } from './Home';
import sinon from 'sinon';
//import configureStore from 'redux-mock-store';
//const mockStore = configureStore();
//const dispatch = sinon.spy();

describe('<Home/>', () => {
  let container;
  const counterValue = 1;
  const handleIncreaseValue = sinon.spy();
  const handleDecreaseValue = sinon.spy();
  const props = {
    handleIncreaseValue,
    handleDecreaseValue,
    counterValue,
  };

  beforeEach(() => {
    container = shallow(<Home { ...props }/>);
  });

  it('Should render home ', () => {
    expect(container).toMatchSnapshot();
  });
  it('Should render home ', () => {
    expect(container.find('h2').length).toEqual(1);
  });
  it('Should call increase function ', () => {
    container = mount(<Home { ...props }/>);
    container.find('button').first().simulate(
      'click',
      {preventDefault() {}}
    )
    expect(handleIncreaseValue.calledOnce).toEqual(true);
  });
  it('Should call increase function ', () => {
    container = mount(<Home { ...props }/>);
    container.find('button').at(1).simulate(
      'click',
      {preventDefault() {}}
    )
    expect(handleDecreaseValue.called).toEqual(true);
  });
});
