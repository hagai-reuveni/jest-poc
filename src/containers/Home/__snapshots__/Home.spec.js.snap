// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home/> Should render home  1`] = `
<div>
  <h2>
    Home Page
  </h2>
  <p>
    The counter value is 
    1
  </p>
  <button
    onClick={[Function]}
  >
    Add
  </button>
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
